src/java/com/puppetlabs/http/client/impl/ClientMetricFilter.java

```diff
@@ -0,0 +1,64 @@
+package com.puppetlabs.http.client.impl;
+
+import com.codahale.metrics.MetricFilter;
+import com.codahale.metrics.Metric;
+import org.apache.commons.lang3.StringUtils;
+
+public class ClientMetricFilter {
```

Comment:
Just curious why you implemented the sub-filters as static classes under a `ClientMetricFilter` class.  Since it doesn't look like the subfilter classes share anything from `ClientMetricFilter`, I think it might look better to have a `metricfilter` namespace under `com.puppetlabs.http.client.impl` and just have these sub-filter classes be standalone classes that live in their own .java source files.  Would that make sense to you?